riscv-executor/src/lib.rs

```diff
@@ -510,8 +697,76 @@ mod builder {
             self.reg_mem.second_last = self.reg_mem.last.clone();
         }
 
-        pub fn finish(self) -> (ExecutionTrace<F>, MemoryState, RegisterMemoryState<F>) {
-            (self.trace, self.mem, self.reg_mem.for_bootloader())
+        pub fn finish(mut self) -> Execution<F> {
+            if let ExecMode::Fast = self.mode {
+                return Execution {
+                    // TODO: in FAST mode we don't extend the trace to next power of two. What should this value be in TRACE/FAST?
+                    main_trace_len: self.len(),
```

Comment:
Answering the comment: I guess this value is fine?

Reply:
changed this so it returns the "actual execution" trace len, before extending to the power of two (which is what the continuations code expects)